### ISSUE TITLE
Style/DocumentDynamicEvalDefinition false positive

### ISSUE THREAD

**Issue by javierav** [rubocop/rubocop#10098]
This code causes `Style/DocumentDynamicEvalDefinition` to throw an error:

```ruby
module Test
  def test(*attributes)
    attributes.each do |attribute|
      class_eval <<~METHOD, __FILE__, __LINE__ + 1
        # def _set_start_at
        #   if @start_at_date && @start_at_time
        #     self.start_at = Time.zone.parse("\#{@start_at_date} \#{@start_at_time}")
        #   end
        # end

        def _set_#{attribute}
          if @#{attribute}_date.present? && @#{attribute}_time.present?
            self.#{attribute} = Time.zone.parse("\#{@#{attribute}_date} \#{@#{attribute}_time}")
          end
        end
      METHOD
    end
  end
end
```

## Expected behavior

I would expect this cop to work with no errors.

## Actual behavior

Rubocop failed with `Style/DocumentDynamicEvalDefinition`.

```
Offenses:

test.rb:4:7: C: Style/DocumentDynamicEvalDefinition: Add a comment block showing its appearance if interpolated.
      class_eval <<~METHOD, __FILE__, __LINE__ + 1
      ^^^^^^^^^^
```

## Steps to reproduce the problem

Run rubocop with `bin/rubocop test.rb`.

## RuboCop version

```
$ bin/rubocop -V
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 2.6.7 x86_64-darwin20)
  - rubocop-performance 1.11.5
  - rubocop-rails 2.12.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.4.0
```

**Comment by koic** [comment]
It seems like the following code comment is missing (`.present?`). Can you apply the following changes?

```diff
      class_eval <<~METHOD, __FILE__, __LINE__ + 1
        # def _set_start_at
-       #   if @start_at_date && @start_at_time
+       #   if @start_at_date.present? && @start_at_time.present?
        #     self.start_at = Time.zone.parse("\#{@start_at_date} \#{@start_at_time}")
        #   end
        # end

        def _set_#{attribute}
          if @#{attribute}_date.present? && @#{attribute}_time.present?
            self.#{attribute} = Time.zone.parse("\#{@#{attribute}_date} \#{@#{attribute}_time}")
          end
        end
      METHOD
```

**Comment by javierav** [comment]
@koic ohh, good catch! I had not detected the error! Sorry and thanks!
